src/components/Search.js

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
+import axios from 'axios';
 
 class Search extends React.Component {
+  state = {
+    query: ''
+  }
+
+  _onChange = e => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value
+    });
+  }
+
+  _onSubmit = (e) => {
+    e.preventDefault();
+    const { query } = this.state;
+    const { onError, onSearch } = this.props;
+
+    axios.get(`http://api.giphy.com/v1/gifs/search?q=${query}&api_key=q7kTB0TwzgVggIoYvFUcT97sfeuEIL53`)
+     .then(response => {
+       onSearch(response.data.data);
+     })
+     .catch(error => {
+       onError(error.message);
+     });
+   }
+
```

Comment:
Moved the search function into search component.  Note the callbacks in the success and catch.